cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/batch/AbstractBatchContext.java

```diff
@@ -92,6 +92,17 @@ public T call() throws Exception {
   }
 
   @Override
+  public <T extends Dataset> T getDataset(final String namespace, final String name)
```

Comment:
BatchContext extends DatasetContext, so the methods have to be defined. Perhaps they should throw UnsupportedOperation? I'd like to see the use cases and user stories, do they ask for cross-namespace access in pipelines/plugins?

One example I could see would be a dataset that is defined in a dedicated namespace. This dataset would be used for lookups by pipelines in other namespaces. In that case, we would need it here.